In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from datetime import date, timedelta
import seaborn as sns
sns.set()

In [2]:
data = pd.read_csv('corona_scrapper/final_data.csv')

In [3]:
data.columns

Index(['CountryName', 'Total Case', 'Total Deaths', 'Total Recovered',
       'Active Cases', 'Critical Cases', 'Total Test', 'Population'],
      dtype='object')

The columns for the dataset are
* `CountryName`: Names of the country
* `Total Cases`: Total cases recorded in a country
* `Total Deaths`: Total deaths recorded in a country
* `Total Recovered`: Total number of patients recovered from COVID-19
* `Active Cases`: Number of active cases in a country
* `Crtical Cases`: Critical cases in a country
* `Total Test`: Total test conducted in a country
* `Population`: Population of a country

In [4]:
data.head()

CountryName Total Case Total Deaths Total Recovered Active Cases  \
0           USA  4,248,759     148,498        2,028,361    2,071,900   
1        Brazil  2,348,200      85,385        1,592,281      670,534   
2         India  1,339,176      31,425          850,303      457,448   
3        Russia    806,720      13,192          597,140      196,388   
4  South Africa    421,996       6,343          245,771      169,882   

  Critical Cases  Total Test     Population  
0         19,098  52,522,208    331,129,317  
1          8,318   4,911,063    212,657,106  
2          8,944  15,849,068  1,380,863,473  
3          2,300  26,610,623    145,938,709  
4            539   2,684,488     59,355,008

This dataset is from 25th of July 2020 and it can be change by the second

In [5]:
data.describe()

CountryName Total Case Total Deaths Total Recovered Active Cases  \
count          213        213          213             209          209   
unique         213        207          144             197          177   
top     Martinique         13                           60            0   
freq             1          2           27               3           15   

       Critical Cases Total Test Population  
count             127        195        209  
unique             75        195        209  
top                 1      5,465  2,543,628  
freq               14          1          1

Judging by the describe method
* There are `213` countries in the dataset
* Total cases has the same number as countries so it doesn't seem right now that it has empty values
* Total deaths also do not have any missing value
* Total recovered seems incomplete
* Active cases also have missing values so maybe they don't have any active case at the moment
* Critical cases have count way less than countries so we it can't be determined right now 
* Total test count is also little less than countries
* Population count compare to country seems to have missing values

In [6]:
data.shape

(213, 8)

In [7]:
data.isna().sum()

CountryName         0
Total Case          0
Total Deaths        0
Total Recovered     4
Active Cases        4
Critical Cases     86
Total Test         18
Population          4
dtype: int64

In [8]:
data.dtypes

CountryName        object
Total Case         object
Total Deaths       object
Total Recovered    object
Active Cases       object
Critical Cases     object
Total Test         object
Population         object
dtype: object

In [9]:
# making a copy of dataframe
df = data

In [10]:
df['Total Case'] = df['Total Case'].str.replace(',','')
df['Total Deaths'] = df['Total Deaths'].str.replace(',','')
df['Total Recovered'] = df['Total Recovered'].str.replace(',','')
df['Active Cases'] = df['Active Cases'].str.replace(',','')
df['Critical Cases'] = df['Critical Cases'].str.replace(',','')
df['Population'] = df['Population'].str.replace(',','')

In [11]:
df[df['Total Deaths']==' ']

CountryName Total Case Total Deaths Total Recovered  \
157                 Vietnam        416                          365   
166                Mongolia        288                          218   
168                 Eritrea        261                          189   
172                Cambodia        202                          142   
173          Faeroe Islands        192                          188   
174               Gibraltar        184                          180   
180              Seychelles        114                           69   
182                  Bhutan         92                           85   
187        French Polynesia         62                           60   
188  St. Vincent Grenadines         52                           39   
191                   Macao         46                           46   
192        Papua New Guinea         32                           11   
194                    Fiji         27                           18   
195             Saint Lucia         24                           22   
196             Timor-Leste         24                           24   
197                 Grenada         23                           23   
198           New Caledonia         22                           22   
199                    Laos         20                           19   
200                Dominica         18                           18   
201   Saint Kitts and Nevis         17                           15   
202        Falkland Islands         13                           13   
203               Greenland         13                           13   
205            Vatican City         12                           12   
206   Caribbean Netherlands         11                            7   
209               St. Barth          6                            6   
210   Saint Pierre Miquelon          4                            1   
211                Anguilla          3                            3   

    Active Cases Critical Cases Total Test Population  
157           51            NaN    275,000   97394394  
166           70              9     32,952    3281467  
168           72            NaN        NaN    3549431  
172           60              1     56,126   16733155  
173            4            NaN     31,481      48875  
174            4            NaN     19,281      33690  
180           45            NaN        NaN      98387  
182            7            NaN     41,075     772141  
187            2            NaN      4,649     281014  
188           13            NaN      1,647     110963  
191            0            NaN      4,071     649879  
192           21            NaN      9,885    8956980  
194            9            NaN      4,000     896865  
195            2            NaN      2,597     183682  
196            0            NaN      1,568    1319919  
197            0            NaN      5,465     112557  
198            0            NaN     10,119     285672  
199            1            NaN     25,121    7281999  
200            0            NaN      1,005      71998  
201            2            NaN        948      53223  
202            0            NaN      1,816       3485  
203            0            NaN      5,645      56777  
205            0            NaN        NaN        801  
206            4            NaN        424      26238  
209            0            NaN        897       9879  
210            3            NaN        NaN       5792  
211            0            NaN        380      15012

In [12]:
df['Total Deaths'] = df['Total Deaths'].replace(' ',0)

In [13]:
df[df['Total Deaths']==' ']

Empty DataFrame
Columns: [CountryName, Total Case, Total Deaths, Total Recovered, Active Cases, Critical Cases, Total Test, Population]
Index: []